posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc

```diff
@@ -0,0 +1,300 @@
+
+Let's first explain what is Serverless?
+
+== Serverless
+Serverless is a short name for Serverless Computing, which is an execution model in which the cloud provider allocates resources on demand. Serverless enables developers to concentrate on their own applications without the need of managing servers. It does not mean there is no server running in the background. On a contrary, normally Serverless architecture contains servers.
+
+Serverless normally executes code in the form of a function, so Serverless sometimes referred to “Functions as a ServICCE” (FaaS). There are a few FaaS from major cloud providers:
+
+*	IBM Cloud Code Engine: IBM Cloud Function
+*	AWS: AWS Lambda
+*	Microsoft Azure: Azure Functions
+*	Google Cloud: Google Cloud Functions
+
+Serverless is a model. In order to work with Serverless, Knative and Cloud Event are often used.  https://knative.dev/docs/concepts/[Knative] provides a platform-agnostic solution for running serverless deployment. CloudEvent is also a common event model to be used in Serverless architecture. This blog will focus on CloudEvent instead.
+
+== CloudEvents
+https://cloudevents.io/[CloudEvents] is a specification of describing event data in a common and language agnostic way. CloudEvents is to simplify event declaration and delivery across the services and platforms. This specification is under Cloud Native Computing Foundation (CNCF). CloudEvents are supported by many Serverless frameworks including IBM Cloud Code Engine, which will be discussed in the next section.
+
+== IBM Cloud Code Engine
+https://www.ibm.com/cloud/code-engine[IBM Cloud Code Engine](ICCE) is a fully managed, serverless platform designed to run container images, batch jobs, or source code. Code Engine allows you to build event driven workloads that react to CloudEvents. Code Engine is truly serverless. It automatically scales your workloads up and down, and even down to zero when there are no requests. You only pay for the resources you consume. You can easily run your cloud native applications in Code Engine. The common cloud native applications normal use MicroProfile specifications for configuring, securing, and observing applications.
+
+== MicroProfile Applications
+MicroProfile Applications are popular cloud native applications. They are designed for containers and clouds. Since we are entering into Serverless era, it will be very important to get MicroProfile Applications running in a serverless environment. Luckily with CloudEvent, it is very straightforward to get these kinds of applications running as functions in an environment that supports CloudEvents such as IBM Cloud Code Engine. In the next section, we will demonstrate how to get MicroProfile application deployed to Open Liberty running as a function in IBM Cloud Code Engine via CloudEvent.
```

Comment:
```suggestion
MicroProfile Applications are popular cloud native applications. MicroProfile offers a set of standards APIs for cloud-native applications that free your applications from vendor lock-in. Open Liberty is the leading implementation for MicroProfile specifications and we will use Open Liberty in this blog. 

Since we are entering into Serverless era, it will be very important to get MicroProfile Applications running in a serverless environment. Luckily with CloudEvent, it is very straightforward to get these kinds of applications running as functions in an environment that supports CloudEvents such as IBM Cloud Code Engine. In the next section, we will demonstrate how to get MicroProfile application deployed to Open Liberty running as a function in IBM Cloud Code Engine via CloudEvent.
```